src/constants.js

```diff
@@ -46,15 +47,21 @@ function getDirectory(category: string): string {
   return path.join(userHome, `.${category}`, 'yarn');
 }
 
-function getCacheDirectory(): string {
+function getPreferredCacheDirectories(): Array<string> {
+  const preferredCacheDirectories = [];
+
   if (process.platform === 'darwin') {
-    return path.join(userHome, 'Library', 'Caches', 'Yarn');
+    preferredCacheDirectories.push(path.join(userHome, 'Library', 'Caches', 'Yarn'));
+  } else {
+    preferredCacheDirectories.push(getDirectory('cache'));
   }
 
-  return getDirectory('cache');
+  preferredCacheDirectories.push(path.join(os.tmpdir(), '.yarn-cache'));
```

Comment:
We should probably append a timestamp or something to the `'.yarn-cache'` name to make it unique across instances and runs? What do you think? (can be a follow-up)